c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -464,6 +464,12 @@ impl TypedAstContext {
         ty.map(|ty| (expr_id, ty))
     }
 
+    pub fn type_for_kind(&self, kind: &CTypeKind) -> Option<CTypeId> {
+        self.c_types
+            .iter()
+            .find_map(|(id, k)| if kind == &k.kind { Some(*id) } else { None })
+    }
```

Comment:
Is it a good idea to be doing an `O(n)` reverse search for this?  Can we just build a reverse `c_type_ids: HashMap<CType, CTypeKind>` once?  Perhaps at the end of `fn ConversionContext::new` or `fn ConversionContext::into_typed_context`?

Reply:
Let's cross the optimization bridge when we come to it.